### ISSUE TITLE
RecursionError when used on subclass of Tensor

### ISSUE THREAD

**Issue by Varal7** [szagoruyko/pytorchviz#65]
When using subclasses of Tensor, `make_dot` crashed with `RecursionError`.

``` python
class DummyTensor(torch.Tensor):
    pass

a = DummyTensor(torch.ones(5, requires_grad=True))
make_dot(a)
```

Output:
```
/usr/local/lib/python3.7/dist-packages/torchviz/dot.py in add_base_tensor(var, color)
    152             dot.edge(str(id(var.grad_fn)), str(id(var)))
    153         if var._is_view():
--> 154             add_base_tensor(var._base, color='darkolivegreen3')
    155             dot.edge(str(id(var._base)), str(id(var)), style="dotted")
    156 

RecursionError: maximum recursion depth exceeded in __instancecheck__
```

The reason is because `a` is a view, so is `a._base`, so is `a._base._base`, so is `a._base._base._base`, etc.

https://colab.research.google.com/drive/1VlAEQY5lOtp3MsYfgyCkrNmkjzuKoRX1?usp=sharing

**Comment by albanD** [comment]
I'm confused, why is a a view here?

**Comment by Varal7** [comment]
¯\_(ツ)_/¯

**Comment by Varal7** [comment]
So the answer is: `a` is a view of `torch.ones(5)`.

This actually seems like a Pytorch bug.

```python
a = torch.ones(1)
b = torch.Tensor(a)
print(b._is_view()) # True
print(b._base is a) # True
```

```python
a = torch.ones(1)
c = DummyTensor(a)
print(c._is_view()) # True
print(c._base is a) # False
print(type(c._base)) # <class '__main__.DummyTensor'>
```

`c._base` here creates a new DummyTensor, as per the [docs](https://pytorch.org/docs/stable/notes/extending.html)
  > As of version 1.7.0, methods on torch.Tensor and functions in public torch.* namespaces applied on torch.Tensor subclasses will return subclass instances instead of torch.Tensor instances:

**Comment by ezyang** [comment]
Yeah, agreed this is a bug.

**Comment by ezyang** [comment]
As a workaround, do the `_base` object within the DisableTorchFunction context manager:

```
with torch._C.DisableTorchFunction():
  base = c._base
```

**Comment by Varal7** [comment]
This is indeed fixed by pytorch/pytorch#60464.

![image](https://user-images.githubusercontent.com/8019486/123996011-e4b31800-d99c-11eb-9614-eb7574d7e9d1.png)
